src/Symfony/Component/DependencyInjection/Compiler/ResolveBindingsPass.php

```diff
@@ -37,8 +37,21 @@ public function process(ContainerBuilder $container)
         try {
             parent::process($container);
 
-            foreach ($this->unusedBindings as list($key, $serviceId)) {
-                $message = sprintf('Unused binding "%s" in service "%s".', $key, $serviceId);
+            foreach ($this->unusedBindings as list($key, $serviceId, $type, $file)) {
+                $message = sprintf('You have a "bind" configured for an argument named "%s" ', $key);
```

Comment:
`for an argument named "%s"`: that part is misleading: binding can also work by type. - and also by type+$name since 4.2. We should adjust the message for that.

Reply:
Done